app/services/migration/migrators/declaration.rb

```diff
@@ -0,0 +1,54 @@
+module Migration::Migrators
+  class Declaration < Base
+    class << self
+      def record_count
+        ecf_declarations.count
+      end
+
+      def model
+        :declaration
+      end
+
+      def ecf_declarations
+        Migration::Ecf::ParticipantDeclaration.includes(:cohort, :declaration_states, cpd_lead_provider: :npq_lead_provider)
+      end
+
+      def dependencies
+        %i[cohort application lead_provider course user]
+      end
+    end
+
+    def call
+      migrate(self.class.ecf_declarations) do |ecf_declaration|
+        declaration = ::Declaration.find_or_initialize_by(ecf_id: ecf_declaration.id)
+        latest_declaration_state = ecf_declaration.declaration_states.order(created_at: :desc).first
```

Comment:
Usually we surface the reason for ineligible states in the serialiser:
```
declaration.declaration_states.detect(&:ineligible?)&.state_reason
```
and we only set this reason in the record declaration if it's ineligible:
```
participant_declaration.make_ineligible!(reason: :duplicate)
```
Maybe we check to only grab the ineligible state reasons similar to the first query and setting it on the declaration?